### PR TITLE
Flashloan Detector Bot Update

### PR DESCRIPTION
**Pull request by Vxatz**
## Changes

- Added AAVE V3 Flashloan ABI
- Added check for failed CoinGecko calls

### CODE REVIEW

File: flashloan-detector/src/helper.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 /* eslint-disable no-plusplus */
-const { ethers, getEthersProvider, getTransactionReceipt } = require('forta-agent');
-const { Contract, Provider } = require('ethers-multicall');
-const axios = require('axios').default;
+const { ethers, getEthersProvider, getTransactionReceipt } = require("forta-agent");
+const { Contract, Provider } = require("ethers-multicall");
```
Comment (RCantu92) [inline flashloan-detector/src/helper.js:3]: Would it be possible to migrate to `MulticallProvider` from the GAM? Similar to how that move was made with the _Asset Drained_ bot.

Comment (RCantu92) [reply flashloan-detector/src/helper.js:3]: Would possibly require an update elsewhere in the files.

File: flashloan-detector/src/helper.js [hunk 2]
```diff
@@ -144,34 +152,40 @@ module.exports = {
...
 
-    const totalTokensProfit = usdTokenProfits
-      .reduce((sum, profit) => sum + profit, 0);
+    const totalTokensProfit = usdTokenProfits.reduce((sum, profit) => sum + profit, 0);
 
     return totalTokensProfit;
   },
   async calculateNativeUsdProfit(amount, token) {
     const response = await axios.get(getNativeTokenPrice(token));
+    if (!response.data[token]) return 0;
```
Comment (RCantu92) [inline flashloan-detector/src/helper.js:173]: This is the equivalent of the `try/catch` method we had initially brought up, correct?

**Review by RCantu92 — COMMENTED** [review]
Approval pending testing with Avalanche endpoint. The following points are also suggestions to implement.
- Let's embed the link into the exploit transaction hash in the `README` as well as adding the two transaction that were provided to us. [`0x0ab12913f9232b27b0664cd2d50e482ad6aa896aeb811b53081712f42d54c026`](https://snowtrace.io/tx/0x0ab12913f9232b27b0664cd2d50e482ad6aa896aeb811b53081712f42d54c026) and [`0xe176bd9cfefd40dc03508e91d856bd1fe72ffc1e9260cd63502db68962b4de1a`](https://bscscan.com/tx/0xe176bd9cfefd40dc03508e91d856bd1fe72ffc1e9260cd63502db68962b4de1a) with their "exploit names" in parenthesis.
- Suggestion to rename `aave-detector.js` to `aave-v2-detector`, since we've now added a `v3` detector

File: flashloan-detector/src/helper.js (hunk 1 above)
Comment (Vxatz) [reply flashloan-detector/src/helper.js:3]: The main difference of the GAM Multicall Provider is that it supports setting the block number. As this bot is not really meant to be executed in old blocks (due to the CG calls), I think that this is not needed in this bot.

Thoughts on this?

File: flashloan-detector/src/helper.js (hunk 2 above)
Comment (Vxatz) [reply flashloan-detector/src/helper.js:173]: Yes, and it's in the same direction as L158.

File: flashloan-detector/src/helper.js (hunk 1 above)
Comment (RCantu92) [reply flashloan-detector/src/helper.js:3]: That's a good point. It can remain.

**Review by RCantu92 — APPROVED** [review]

**Pull request merged**